private/functions/New-DbaLogShippingSecondaryPrimary.ps1

```diff
@@ -131,7 +132,7 @@ function New-DbaLogShippingSecondaryPrimary {
         Stop-Function -Message "The backup destination path should be formatted in the form \\server\share." -Target $SqlInstance
         return
     } else {
-        if (-not ((Test-Path $BackupDestinationDirectory -PathType Container -IsValid) -and ((Get-Item $BackupDestinationDirectory).PSProvider.Name -eq 'FileSystem'))) {
+        if (-not ((Test-DbaPath $BackupDestinationDirectory -SqlInstance $SqlInstance -SqlCredential $SqlCredential) -and ((Get-Item $BackupDestinationDirectory).PSProvider.Name -eq 'FileSystem'))) {
```

Comment:
```suggestion
        if (-not ((Test-DbaPath $BackupDestinationDirectory -SqlInstance $ServerSecondary) -and ((Get-Item $BackupDestinationDirectory).PSProvider.Name -eq 'FileSystem'))) {
```

Please reuse connections already created via Connect call. This also makes it clear we are checking the path on the secondary server.